src/datepicker/datepicker-config.ts

```diff
@@ -23,5 +22,5 @@ export class NgbDatepickerConfig {
 	outsideDays: 'visible' | 'collapsed' | 'hidden' = 'visible';
 	showWeekNumbers = false;
 	startDate: { year: number; month: number; day?: number };
-	weekdays: TranslationWidth | boolean = TranslationWidth.Short;
+	weekdays: 'short' | 'long' | 'narrow' | boolean = 'narrow';
```

Comment:
I was wondering about this change.

1. If there is any value in extracting the type from the `DateFormatOptions`, ex. like so:

`Exclude<Intl.DateTimeFormatOptions['weekday'], undefined>`

I don't think I like it though, just strings would do I guess...

2. We have `static ngAcceptInputType_weekdays: boolean | number;` in the `NgbDatepicker`, I guess it should be changed to accept `boolean | number` as well.

https://stackblitz.com/edit/angular-ubkevf-uk9k3f?file=src%2Fapp%2Fdatepicker-basic.html

4. As there is no compatible `Intl` formatting option with `TranslationWidth.Short` (closest one being 'narrow'), display changes ex. from `Tu` → `T`?

---

So in the end it is a breaking change with:

Previously:

```html
<ngb-datepicker [weekdays]="2">
```

```ts
this.weekdays = TranslationWidth.Short;
```

Displayed `Tuesday` as `Tu`

Now:
```html
<ngb-datepicker weekdays='short'>
```

```ts
this.weekdays = 'short';
```

Displaying `Tuesday` as `T`

Right?

<img width="255" alt="Screenshot 2024-05-30 at 11 32 33" src="https://github.com/ng-bootstrap/ng-bootstrap/assets/8074436/e929c185-6d14-4b16-a61e-2bded9b5dc37">

Reply:
Didn't know about the exclude, it's nice :)

I didn't know what was the purpose of `ngAcceptInputType_weekdays` and after some digging (https://blog.ninja-squad.com/2021/11/03/what-is-new-angular-13.0/) it's not needed anymore since Angular 13. So I will remove them if it's ok?

And yes there is no perfect equivalent to TranslationWidth.Short in Intl...

Reply:
If `ngAcceptInputType_xxx` is not required anymore, we should remove it everywhere, so a separate PR would be better I guess for tracking/history purposes. Th blog lists only getters/setters though, @jnizet would know better, it's their article ;)

Reply:
OK will do the change to accept string or boolean for now.

Lots of guys from France ;)